partiql-ast/src/main/java/org/partiql/ast/literal/LiteralTypedString.java

```diff
@@ -0,0 +1,34 @@
+package org.partiql.ast.literal;
+
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.DataType;
+
+/**
+ * TODO docs
+ * Represent type + 'some string value'
+ */
+@EqualsAndHashCode(callSuper = false)
+public class LiteralTypedString extends Literal {
+    @NotNull
+    public DataType type;
+
+    @NotNull
+    public String value;
+
+    private LiteralTypedString(@NotNull DataType type, @NotNull String value) {
+        this.type = type;
+        this.value = value;
+    }
+
+    @NotNull
+    public static LiteralTypedString litTypedString(@NotNull DataType type, @NotNull String value) {
+        return new LiteralTypedString(type, value);
+    }
```

Comment:
> To make sure people don't put in bad types, I'd consider making this a final class and then exchanging this method for litTime(int precision, @NotNull String value), litTimeZ(int precision, @NotNull String value), and so on and so forth.

I like the idea to create date/time/timestamp-specific factory methods. Perhaps I could add those after I get the general design approved first? We could always add these in a subsequent PR or release.

---
> Or, just making them their own classes (LiteralTime, LiteralTimestamp, LiteralDate). I imagine it would have the same benefits that you described in [#1650 (files)](https://github.com/partiql/partiql-lang-kotlin/pull/1650/files#r1851189538), and it would more closely match the EBNF and more straightforward.

For now, I liked the modeling of these `<data type > '<string repr in quotes>'` to be under one class and reuse the data type enum. The parser does perform some validation to confirm that the strings match the EBNF rules.